nomad/server.go

```diff
@@ -1665,6 +1670,44 @@ func (s *Server) stopOldWorkers(oldWorkers []*Worker) {
 	}
 }
 
+// listenWorkerEvents listens for events emitted by scheduler workers and log
+// them if necessary. Some events may be skipped to avoid polluting logs with
+// duplicates.
+func (s *Server) listenWorkerEvents() {
+	loggedAt := make(map[string]time.Time)
+
+	gcDeadline := 4 * time.Hour
+	gcTicker := time.NewTicker(10 * time.Second)
+	defer gcTicker.Stop()
+
+	for {
+		select {
+		case <-gcTicker.C:
+			for k, v := range loggedAt {
+				if time.Since(v) >= gcDeadline {
+					delete(loggedAt, k)
+				}
+			}
+		case e := <-s.workersEventCh:
+			switch event := e.(type) {
+			case *scheduler.PortCollisionEvent:
+				if event != nil && event.Node != nil {
+					if _, ok := loggedAt[event.Node.ID]; !ok {
```

Comment:
No need to change but always feel free to save yourself an indent by making the condition for the skip case.

```suggestion
					if _, ok := loggedAt[event.Node.ID]; ok { continue }
```

Reply:
Looks much nicer now, thanks!